google_guest_agent/addresses.go

```diff
@@ -85,16 +85,24 @@ func getForwardsFromRegistry(mac string) ([]string, error) {
 	return regFwdIPs, nil
 }
 
-func compareIPs(configuredIPs, desiredIPs []string) (toAdd, toRm []string) {
-	for _, desiredIP := range desiredIPs {
-		if !containsString(desiredIP, configuredIPs) {
-			toAdd = append(toAdd, desiredIP)
+func compareRoutes(configuredRoutes, desiredRoutes []string) (toAdd, toRm []string) {
+	for _, desiredRoute := range desiredRoutes {
+		if !strings.Contains(desiredRoute, "/") {
+			logger.Errorf("requested route %q missing mask, skipping.", desiredRoute)
+			continue
+		}
```

Comment:
As opposed to dropping the desiredRoute, I'd suggest just appending /32. 

We talked offline, and it sounds like your desired approach is to update all of the callers to ensure they have the mask applied to each member. This case, then, should never happen.  By adding /32 here, though, you'll at least be able to recover instead of dropping the route.

Reply:
yeah, i don't think the compare function should modify its inputs at all. we take from metadata and from `getLocalRoutes()`, the latter of which is patched in this PR to append /32 to entries without any mask.